homeassistant/components/shelly/climate.py

```diff
@@ -316,5 +316,5 @@ def _handle_coordinator_update(self) -> None:
                 self.block = block
 
                 _LOGGER.debug("Entity %s attached to block", self.name)
-                self.async_write_ha_state()
-                return
+
+        self.async_write_ha_state()
```

Comment:
Why is it out of the loop? If device is not initialized we already call update (see line 307), when we search for a block we should not call the update method if there is a problem.
Also this needs to be `super()._update_callback()`

Reply:
> Why is it out of the loop?

Why do you want it in the loop? It is enough to call it once.

> Also this needs to be super()._update_callback()

`super()` object doesn't have `_update_callback()` method.